web/modules/custom/hoeringsportal_audit_log/src/Helpers/ConfigHelper.php

```diff
@@ -89,11 +89,33 @@ public function escapeProviderId(string $input) {
   /**
    * Get route names.
    *
-   * @return array<string, string>
-   *   Array of route names.
+   * @return array<string, string>|null
+   *   Array of route names or NULL.
    */
-  public function getRouteNames() {
-    return $this->moduleConfig->get('logged_route_names');
+  public function getRouteNames() : ?array {
+    $routesToLog = $this->moduleConfig->get('routes_to_log');
+    if ($routesToLog && array_key_exists('routes', $routesToLog)) {
+      return $routesToLog['routes'];
+    }
+    else {
+      return NULL;
+    }
```

Comment:
```suggestion
    return (array) $this->moduleConfig->get('routes_to_log')['routes'] ?? NULL;
```

Casting to an `array` works magically in PHP (https://www.php.net/manual/en/language.types.array.php#language.types.array.casting)